File: backend/app/schemas/answer.py
from pydantic import BaseModel
from typing import Any, Optional
from datetime import datetime

class AnswerBase(BaseModel):
    question_id: int
    student_answer: Optional[str] = None

class AnswerCreate(AnswerBase):
    pass

class AnswerUpdate(BaseModel):
    student_answer: Optional[str] = None
    score: Optional[int] = None
    is_correct: Optional[bool] = None

class AnswerResponse(AnswerBase):
    id: int
    record_id: int
    score: int
    is_correct: Optional[bool] = None
    graded_at: Optional[datetime] = None
    grader_id: Optional[int] = None
    created_at: datetime

    class Config:
        from_attributes = True

class GradeRequest(BaseModel):
    score: int
    is_correct: Optional[bool] = None
    override_reason: Optional[str] = None


class AiGradingResponse(BaseModel):
    answer_id: int
    question_id: int
    record_id: int
    grading_status: str
    grading_source: str
    ai_score: Optional[int] = None
    ai_feedback: Optional[dict[str, Any]] = None
    ai_model: Optional[str] = None
    ai_graded_at: Optional[datetime] = None
    last_error: Optional[str] = None

File: frontend/src/types/answer.ts
import type { QuestionType } from './question';

// 阅卷接口嵌套返回的题目信息（仅字段子集）
export type GradingQuestion = {
  type: QuestionType;
  content: string;
  options?: string[] | null;
  answer?: string | null;
  score: number;
};

export interface AiGrading {
  answer_id: number;
  question_id: number;
  record_id: number;
  grading_status: 'pending' | 'processing' | 'completed' | 'failed' | string;
  grading_source: 'pending' | 'ai' | 'teacher' | 'failed' | string;
  ai_score?: number | null;
  ai_feedback?: {
    reasoning?: string;
    confidence?: number;
    criterion_results?: Array<{ criterion_id: string; score: number; reason: string }>;
  } | null;
  ai_model?: string | null;
  ai_graded_at?: string | null;
  last_error?: string | null;
}

export interface Answer {
  id: number;
  record_id: number;
  question_id: number;
  student_answer?: string | null;
  score: number;
  is_correct?: boolean | null;
  graded_at?: string | null;
  grader_id?: number | null;
  created_at: string;
  // 阅卷接口嵌套返回题目信息（字段子集）
  question?: GradingQuestion;
  ai_grading?: AiGrading;
}

export interface GradeRequest {
  score: number;
  is_correct?: boolean;
  override_reason?: string;
}

export type AnswerValue = string | string[];
